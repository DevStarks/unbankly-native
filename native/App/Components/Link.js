import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Text, TouchableOpacity } from 'react-native'
import styles from './Styles/LinkStyles'
import ExamplesRegistry from '../Services/ExamplesRegistry'

// Ignore in coverage report
/* istanbul ignore next */
ExamplesRegistry.addComponentExample('Link', () =>
  <Link onPress={() => window.alert('Your drawers are showing')}>
    This is a link
  </Link>
)

class Link extends Component {
  static propTypes = {
    onPress: PropTypes.func.isRequired,
    children: PropTypes.string.isRequired
  }

  render () {
    return (
      <TouchableOpacity style={[styles.link, this.props.style]} onPress={this.props.onPress}>
        <Text style={styles.link}>{this.props.children}</Text>
      </TouchableOpacity>
    )
  }
}

export default Link
