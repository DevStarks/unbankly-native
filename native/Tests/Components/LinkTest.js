import 'react-native'
import React from 'react'
import Link from '../../App/Components/Link'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

test('Link component renders correctly', () => {
  const tree = renderer.create(<Link onPress={() => {}}>This is a link</Link>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('onPress', () => {
  let i = 0 // i guess i could have used sinon here too... less is more i guess
  const onPress = () => i++
  const wrapperPress = shallow(<Link onPress={onPress}>Link</Link>)

  expect(wrapperPress.prop('onPress')).toBe(onPress) // uses the right handler
  expect(i).toBe(0)
  wrapperPress.simulate('press')
  expect(i).toBe(1)
})
