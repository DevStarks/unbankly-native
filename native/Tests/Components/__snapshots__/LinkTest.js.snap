// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "color": "#637381",
      "fontFamily": "AmaticSC-Bold",
      "fontSize": 17,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#637381",
        "fontFamily": "AmaticSC-Bold",
        "fontSize": 17,
      }
    }
  >
    This is a link
  </Text>
</View>
`;
